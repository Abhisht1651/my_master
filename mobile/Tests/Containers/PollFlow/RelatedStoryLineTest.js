// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import RelatedStoryLineScreen from '../../../App/Containers/Poll/RelatedStoryLines'
import StoryLineDefault from '../../../App/Components/StoryLineShortCardDefualt'

describe('RelatedStoryLine', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<RelatedStoryLineScreen />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<RelatedStoryLineScreen />)
    expect(wrapper.find('Text')).toHaveLength(1)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<RelatedStoryLineScreen />)
    expect(wrapper.find('Image')).toHaveLength(1)
  })
  it('should render the StoryLineDefault Component', () => {
    const wrapper = shallow(<RelatedStoryLineScreen />)
    expect(wrapper.find(StoryLineDefault)).toHaveLength(3)
  })
})
