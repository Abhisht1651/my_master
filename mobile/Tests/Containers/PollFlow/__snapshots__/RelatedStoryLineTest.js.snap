// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedStoryLine should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 44,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": 122,
        "width": 750,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 34,
            "marginLeft": 37.5,
            "width": 44,
          }
        }
      />
    </ForwardRef>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 40,
          "letterSpacing": -0.08,
          "marginLeft": 142.5,
          "opacity": 1,
        }
      }
    >
      Related StoryLines
    </Text>
  </View>
  <ScrollViewMock
    showsVerticalScrollIndicator={false}
  >
    <View
      style={
        Object {
          "marginTop": 90.5,
        }
      }
    >
      <View
        key="0"
        style={
          Object {
            "height": 644,
            "marginBottom": 94.5,
            "width": 600,
          }
        }
      >
        <storyLineShortCardDefault
          border={true}
          followDisplay={true}
          followerDisplay={true}
          following={true}
          friendsFollowing="45 Followers"
          heading="World"
          onPressFollow={[Function]}
          reach={45}
          storyImage={
            Object {
              "default": "",
            }
          }
          storyTitle="UK exit from the European Union"
          trending={true}
          updatedTime="2m"
        />
      </View>
      <View
        key="1"
        style={
          Object {
            "height": 644,
            "marginBottom": 94.5,
            "width": 600,
          }
        }
      >
        <storyLineShortCardDefault
          border={true}
          followDisplay={true}
          followerDisplay={true}
          following={true}
          friendsFollowing="45 Followers"
          heading="World"
          onPressFollow={[Function]}
          reach={45}
          storyImage={
            Object {
              "default": "",
            }
          }
          storyTitle="UK exit from the European Union"
          trending={true}
          updatedTime="2m"
        />
      </View>
      <View
        key="2"
        style={
          Object {
            "height": 644,
            "marginBottom": 94.5,
            "width": 600,
          }
        }
      >
        <storyLineShortCardDefault
          border={true}
          followDisplay={true}
          followerDisplay={true}
          following={true}
          friendsFollowing="45 Followers"
          heading="World"
          onPressFollow={[Function]}
          reach={45}
          storyImage={
            Object {
              "default": "",
            }
          }
          storyTitle="UK exit from the European Union"
          trending={true}
          updatedTime="2m"
        />
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
